admin/class-yoast-form.php

```diff
@@ -0,0 +1,419 @@
+			settings_fields( $option_long_name );
+		}
+		$this->set_option( $option );
+	}
+
+	/**
+	 * Set the option used in output for form elements
+	 *
+	 * @param $option
+	 */
+	public function set_option( $option ) {
+		$this->option  = $option;
+		$this->options = $this->get_option( $option );
+	}
+
+	/**
+	 * Retrieve options based on whether we're on multisite or not.
+	 *
+	 * @since 1.2.4
+	 *
+	 * @param string $option The option to retrieve.
+	 *
+	 * @return array
+	 */
+	private function get_option( $option ) {
+		if ( is_network_admin() ) {
+			return get_site_option( $option );
+		} else {
+			return get_option( $option );
+		}
+	}
+
+	/**
+	 * Generates the footer for admin pages
+	 *
+	 * @param bool $submit       Whether or not a submit button and form end tag should be shown.
+	 * @param bool $show_sidebar Whether or not to show the banner sidebar - used by premium plugins to disable it
+	 */
+	public function admin_footer( $submit = true, $show_sidebar = true ) {
+		if ( $submit ) {
+			submit_button();
+
+			echo '
+			</form>';
+		}
+
+		do_action( 'wpseo_admin_footer' );
+
+		echo '
+			</div><!-- end of div wpseo_content_top -->';
+
+		if ( $show_sidebar ) {
+			$this->admin_sidebar();
+		}
+
+		echo '</div><!-- end of div wpseo_content_wrapper -->';
+
+
+		if ( ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) ) {
+			$xdebug = ( extension_loaded( 'xdebug' ) ? true : false );
+			echo '
+			<div id="poststuff">
+			<div id="wpseo-debug-info" class="postbox">
+
+				<h3 class="hndle"><span>' . __( 'Debug Information', 'wordpress-seo' ) . '</span></h3>
+				<div class="inside">
+					<h4>' . esc_html( __( 'Current option:', 'wordpress-seo' ) ) . ' <span class="wpseo-debug">' . esc_html( $this->option ) . '</span></h4>
+					' . ( $xdebug ? '' : '<pre>' );
+			var_dump( $this->get_option( $this->option ) );
+			echo '
+					' . ( $xdebug ? '' : '</pre>' ) . '
+				</div>
+			</div>
+			</div>';
+		}
+
+		echo '
+			</div><!-- end of wrap -->';
+	}
+
+	/**
+	 * Generates the sidebar for admin pages.
+	 */
+	public function admin_sidebar() {
+
+		// No banners in Premium
+		if ( class_exists( 'WPSEO_Product_Premium' ) ) {
+			$license_manager = new Yoast_Plugin_License_Manager( new WPSEO_Product_Premium() );
+			if ( $license_manager->license_is_valid() ) {
+				return;
+			}
+		}
+
+		$service_banners = array(
+			array(
+				'url' => 'https://yoast.com/hire-us/website-review/#utm_source=wordpress-seo-config&utm_medium=banner&utm_campaign=website-review-banner',
+				'img' => 'banner-website-review.png',
+				'alt' => 'Website Review banner',
+			),
+		);
+
+		$plugin_banners = array(
+			array(
+				'url' => 'https://yoast.com/wordpress/plugins/seo-premium/#utm_source=wordpress-seo-config&utm_medium=banner&utm_campaign=premium-seo-banner',
+				'img' => 'banner-premium-seo.png',
+				'alt' => 'Banner WordPress SEO Premium',
+			),
+		);
+
+		if ( ! class_exists( 'wpseo_Video_Sitemap' ) ) {
```

Comment:
I should create a filter to allow other Yoast plugins to extend this array, but this may be a nice feature for the future.